### PR TITLE
Fixes #31469 - skip cv puppet envs on SP sync history

### PR DESCRIPTION
**Pull request by jlsherrill**

### CODE REVIEW

**Comment by jlsherrill** [comment]
This does 2 things:

* better handles errors during sync proxy history event creation
* does not create history events for CV puppet repos

File: app/lib/actions/middleware/record_smart_proxy_sync_history.rb [hunk 1]
```diff
@@ -1,12 +1,32 @@
...
+          repo_id = action.input[:repository_id]
+          repo = ::Katello::Repository.find_by(id: repo_id)
           smart_proxy_id = action.input[:smart_proxy_id] || action.input[:capsule_id]
-          smart_proxy = ::SmartProxy.find(smart_proxy_id)
-          self.action.output[:smart_proxy_history_id] = repo.create_smart_proxy_sync_history(smart_proxy)
+          smart_proxy = ::SmartProxy.find_by(id: smart_proxy_id)
+
+          #skip pulp2 puppet syncs
+          if (repo_pulp_id = action.input[:repo_pulp_id])
+            return if Katello::ContentViewPuppetEnvironment.find_by(pulp_id: repo_pulp_id)
```
Comment (sjha4) [inline app/lib/actions/middleware/record_smart_proxy_sync_history.rb:13]: It seems to be failing without namespacing..
```suggestion
            return if ::Katello::ContentViewPuppetEnvironment.find_by(pulp_id: repo_pulp_id)
```

Comment (jlsherrill) [reply app/lib/actions/middleware/record_smart_proxy_sync_history.rb]: yeah, load order can cause issues on this randomly, fixed!

**Review by sjha4 — APPROVED** [review]
[test katello]

APJ

**Comment by evgeni** [comment]
@jlsherrill @sjha4 the test failures are JS related, so I'd say not from this PR? :)

**Comment by jlsherrill** [comment]
[test katello]

**Pull request merged**